controller/suite_test.go

```diff
@@ -52,12 +59,49 @@ func TestControllers(t *testing.T) {
 	RunSpecs(t, "Controller Suite")
 }
 
+func getFilePathToCAPICRDs() string {
+	modBits, err := os.ReadFile(filepath.Join("..", "go.mod"))
+	if err != nil {
+		return ""
+	}
+
+	var clusterAPIVersion string
+	for _, line := range strings.Split(string(modBits), "\n") {
+		matches := clusterAPIVersionRegex.FindStringSubmatch(line)
+		if len(matches) == 3 {
+			clusterAPIVersion = matches[2]
+		}
+	}
+
+	if clusterAPIVersion == "" {
+		return ""
+	}
+
+	gopath := envOr("GOPATH", build.Default.GOPATH)
+	return filepath.Join(gopath, "pkg", "mod", "sigs.k8s.io", fmt.Sprintf("cluster-api@v%s", clusterAPIVersion), "config", "crd", "bases")
+}
+
+func envOr(envKey, defaultValue string) string {
+	if value, ok := os.LookupEnv(envKey); ok {
+		return value
+	}
+	return defaultValue
+}
```

Comment:
Should we just hardcode the directory i.e. `cluster-api@v1.6.3`? It'd be easy enough to change when we upgrade cluster-api in go.mod.

Reply:
Yeah we can certainly do that I have no strong opinions either way. Just to make sure we are clear, you are proposing we do something like this?
```
func getFilePathToCAPICRDs() string {
	gopath := os.LookupEnv("GOPATH", build.Default.GOPATH)
	return filepath.Join(gopath, "pkg", "mod", "sigs.k8s.io", "cluster-api@v1.6.3", "config", "crd", "bases")
}
```

Reply:
Maybe
```
f, err := modfile.Parse("go.mod", file_bytes, nil)
if err != nil {
	panic(err)
}

for _, s := range f.Require {
	if strings.Contains(string(s.Mod.Path), "cluster-api") {
		clusterAPIVersion = s.Mod.Version
	}
}
```
and then
```
filepath.Join(gopath, "pkg", "mod", "sigs.k8s.io", fmt.Sprintf("cluster-api@%s", clusterAPIVersion), "config", "crd", "bases")
```

Reply:
https://github.com/linode/cluster-api-provider-linode/pull/247/commits/c4424e6f98e1f985760d278285c9b37b89fbc20b

Reply:
This approach LGTM!